doc/developer_guide/coding-style.rst

```diff
@@ -114,6 +114,27 @@ Additional rules that apply:
   #) Any new line of code must be covered by at least one test case,
      see :ref:`test_suite` and especially :ref:`test_coverage`.
 
+Exceptions
+----------
+
+When raising an exception, every effort must be made to ensure that
+the associated message is clear and provides as much information as
+possible. However, it should *not* contain the name of the current
+routine/class etc. as this is provided in the Python stacktrace.
+Where it makes sense to do so, the past tense should be used,
+e.g. "expected a str but got an object of type 'blah'."
+
+In the event that code that is handling an exception
+then needs to raise a new exception, 'raise from' should be used so as
```

Comment:
I suggest adding an example of `raise from`, similar to how it is done for `six` package below.

Reply:
The point is that we can't use 'raise from' as it's Python 3 only but obviously I hadn't explained that very well :-) I've tried again...